docs/proposals/autoscaling.md

```diff
@@ -197,6 +178,34 @@ resolves to a value that can be checked against a configured threshold.
 Of note: If the statistics gathering mechanisms can be initialized with a registry other components storing statistics can
 potentially piggyback on this registry.
 
+### Interactions with a deployment
+
+In a deployment it is likely that multiple replication controllers must be monitored.  For instance, in a [rolling deployment](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/replication-controller.md#rolling-updates)
+there will be multiple replication controllers, with one scaling up and another scaling down.  This means that an 
+auto-scaler must be aware of the entire set of capacity that backs a service so it does not fight with the deployer.  `AutoScalerSpec.Selector` 
+is what provides this ability.  By using a selector that spans the entire service the auto-scaler can monitor capacity 
+of multiple replication controllers and check that capacity against the `AutoScalerSpec.MaxAutoScaleCount` and 
+`AutoScalerSpec.MinAutoScaleCount` while still only targeting a specific `ReplicationController`.  
+
+In the course of a rolling update it would be up to the deployment orchestration to re-target the auto-scaler.  
```

Comment:
This is going to be somewhat problematic - how do you know which auto-scalers target you?  How do you transactionally guarantee that?  It also means every component that creates and manipulates replication controllers also has to know about auto-scalers.  That's not necessarily a blocking issue, but it does create complexity when it comes to meaning.  It's possible that the deployment process itself is the thing that should be taking the target of the autoscaler (via the resize verb), rather than the underlying individual replication controllers.  In the OpenShift case, we have a deployment target (the deployment config) which can support resize.  In the current Kube code, kubectl has no target, so it would have to talk to the controller instead.

It seems like the deployer really needs a way to communicate to the autoscaler that is tied to the replication controllers directly.  If you had that, you could mark the replication controllers semantically as you went through a deployment.  However, that has some issues too.

Basically, kubectl having to know about auto-scalers to do its job feels wrong. We need to establish whether it's impossible to keep kubectl isolated from the auto-scaler, or whether we just need a better abstraction (some combination of a marker on the RC + the autoscaler selector).